### PR TITLE
Gumballs and gumball jars

### PR DESCRIPTION
**Pull request by Hockaa**
Adds gumballs (a regular and a medical variant) to the game. Normal variants initialize similarly to normal chewing gum. Medical gumballs are the same except they also have 5u of Tricordrazine in them. A variant of glass jar was added that spawn full of gumballs (currently 15). There are both a regular and medical variant of the gumball jars. A medical gumball jar now spawns on a table in Medical at roundstart. The regular variant has a chance to spawn in cargo.

The damage_per_crunch var and it's subsequent functionality were added as per Matt's request and works for all chewable items. It is always an empty var by default.

### CODE REVIEW

**Comment by Hockaa** [comment]
!review

File: code/game/objects/items/glassjar.dm [hunk 1]
```diff
@@ -12,14 +19,20 @@
...
+	var/list/gumballs_contained = list()
 	drop_sound = 'sound/items/drop/glass.ogg'
 	pickup_sound = 'sound/items/pickup/glass.ogg'
 
 /obj/item/glass_jar/New()
 	..()
 	update_icon()
 
+/obj/item/glass_jar/MouseDrop(atom/target)
+	if(ishuman(target) || issmall(target) && target == usr)
```
Comment (Geevies) [inline code/game/objects/items/glassjar.dm:32]: check adjacency, use_check_and_message maybe

File: code/game/objects/items/glassjar.dm [hunk 2]
```diff
@@ -120,13 +136,61 @@
...
 	return
 
+/obj/item/glass_jar/proc/handle_gumball_addition(mob/user, var/obj/item/clothing/mask/chewable/candy/gum/gumball/G) // handles the entire gumbball addition process
+	if(length(gumballs_contained) < GUMBALL_MAX)
+		gumballs_contained += G
+		user.drop_from_inventory(G)
+		G.forceMove(src)
+		if(!contains)
+			contains = JAR_GUMBALL
+		user.visible_message("[user] puts a gumball in \the [name].", SPAN_NOTICE("You put a gumball in \the [name]."))
```
Comment (Geevies) [inline code/game/objects/items/glassjar.dm:161]:
```suggestion
		user.visible_message("<b>[user]</b> puts a gumball in \the [src].", SPAN_NOTICE("You put a gumball in \the [src]."))
```

File: code/game/objects/items/glassjar.dm [hunk 3]
```diff
@@ -120,13 +136,61 @@
...
+		handle_gumball_underlays()
+	else
+		to_chat(user, SPAN_WARNING("\The [name] is full!"))
+
+/obj/item/glass_jar/proc/handle_gumball_removal(mob/user) // same as the addition proc but we're removing gum instead
+	if(length(gumballs_contained))
+		user.put_in_hands(gumballs_contained[1])
+		gumballs_contained -= gumballs_contained[1]
+		user.visible_message("[user] takes a gumball from \the [name].", SPAN_NOTICE("You take a gumball from \the [name]."))
```
Comment (Geevies) [inline code/game/objects/items/glassjar.dm:170]:
```suggestion
		user.visible_message("<b>[user]</b> takes a gumball from \the [src].", SPAN_NOTICE("You take a gumball from \the [src]."))
```

File: code/game/objects/items/glassjar.dm [hunk 4]
```diff
@@ -120,13 +136,61 @@
...
+				gumballs_overlay.icon_state = "gumball_max"
+		underlays += gumballs_overlay
+
+
 /obj/item/glass_jar/peter/
 	name = "Peter's Jar"
 
 /obj/item/glass_jar/peter/Initialize()
-	. = ..()
+	..()
```
Comment (Geevies) [inline code/game/objects/items/glassjar.dm:193]: you never add a return hint here, so it has to be `. = ..()` to return the old one

File: code/game/objects/items/glassjar.dm [hunk 5]
```diff
@@ -135,7 +199,35 @@
...
+		var/obj/item/clothing/mask/chewable/candy/gum/gumball/G = new
+		G.forceMove(src)
+		gumballs_contained += G
+
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/item/glass_jar/gumball/LateInitialize()
+	update_icon()
+
+/obj/item/glass_jar/gumball/medical
```
Comment (Geevies) [inline code/game/objects/items/glassjar.dm:217]: you can delete this line, it'll work fine still

File: code/game/objects/items/glassjar.dm [hunk 6]
```diff
@@ -135,7 +199,35 @@
...
+
+/obj/item/glass_jar/gumball/medical/Initialize()
+	for(var/i = 1 to GUMBALL_MAX)
```
Comment (Geevies) [inline code/game/objects/items/glassjar.dm:220]:
```suggestion
/obj/item/glass_jar/gumball/medical/Initialize()
	. = ..()
	for(var/i = 1 to GUMBALL_MAX)
```

File: code/game/objects/items/glassjar.dm [hunk 7]
```diff
@@ -135,7 +199,35 @@
...
 	update_icon()
 
+/obj/item/glass_jar/gumball
+	contains = JAR_GUMBALL
+
+/obj/item/glass_jar/gumball/Initialize()
+	..()
+	for(var/i = 1 to GUMBALL_MAX)
+		var/obj/item/clothing/mask/chewable/candy/gum/gumball/G = new
+		G.forceMove(src)
```
Comment (Geevies) [inline code/game/objects/items/glassjar.dm:209]:
```suggestion
		var/obj/item/clothing/mask/chewable/candy/gum/gumball/G = new(src)
```

File: code/game/objects/items/glassjar.dm [hunk 8]
```diff
@@ -135,7 +199,35 @@
...
+
+/obj/item/glass_jar/gumball/LateInitialize()
+	update_icon()
+
+/obj/item/glass_jar/gumball/medical
+
+/obj/item/glass_jar/gumball/medical/Initialize()
+	for(var/i = 1 to GUMBALL_MAX)
+		var/obj/item/clothing/mask/chewable/candy/gum/gumball/medical/G = new
+		G.forceMove(src)
```
Comment (Geevies) [inline code/game/objects/items/glassjar.dm:222]:
```suggestion
		var/obj/item/clothing/mask/chewable/candy/gum/gumball/medical/G = new(src)
```

File: code/game/objects/items/weapons/chewables.dm [hunk 9]
```diff
@@ -167,6 +180,20 @@ obj/item/clothing/mask/chewable/Destroy()
...
 	update_icon()
 
+/obj/item/clothing/mask/chewable/candy/gum/gumball
+	name = "gumball"
+	desc = "A gumball, created and patented by Chip Getmore. Known to contain a hard shell and a reagent interior!"
+	icon_state = "gumball"
+	item_state = null
+	wrapped = FALSE
+
+/obj/item/clothing/mask/chewable/candy/gum/gumball/medical
```
Comment (Geevies) [inline code/game/objects/items/weapons/chewables.dm:190]: you can delete this line too

**Review by Geevies — APPROVED** [review]

File: code/game/objects/items/glassjar.dm (hunk 6 above)
Comment (Hockaa) [reply code/game/objects/items/glassjar.dm]: This actually causes the medical jar to spawn with 15 of the normal gumballs at the same time.

**Review by Chada-1 — APPROVED** [review]

**Pull request merged**